numpy/core/src/multiarray/common.h

```diff
@@ -169,6 +179,11 @@ check_and_adjust_axis(int *axis, int ndim)
     }
     return 0;
 }
+static NPY_INLINE int
+check_and_adjust_axis(int *axis, int ndim)
+{
+    return check_and_adjust_axis_msg(axis, ndim, Py_None);
```

Comment:
> Using NULL seems cleaner though.

Why? We already need to handle `Py_None` being passed in from user code. Furthermore, when we get `Py_None` from usercode, it's a borrowed reference - so like here, it doesn't need an incref unless it escapes to python code.